src/test/java/io/jenkins/plugins/analysis/warnings/DockerITest.java

```diff
@@ -0,0 +1,133 @@
+ */
+public class DockerITest extends IntegrationTestWithJenkinsPerSuite {
+
+    /**
+     * Docker container supporting jenkins agents and gcc/make projects.
+     */
+    @Rule
+    public DockerRule<GccMakeContainer> gccDockerRule = new DockerRule<>(GccMakeContainer.class);
+
+    /**
+     * Docker container supporting jenkins agents and java projects.
+     */
+    @Rule
+    public DockerRule<JavaContainer> javaDockerRule = new DockerRule<>(JavaContainer.class);
+
+    /**
+     * Checks if the Analysis result is correctly, when running inside a java container.
+     *
+     * @throws Exception
+     *         When slave creation or waiting for node start fails.
+     */
+    @Test
+    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
+    public void checkRunningInJavaContainerDocker() throws Exception {
+        DumbSlave agent = createAgent(javaDockerRule.get());
+        WorkflowJob job = createPipeline();
+        job.setDefinition(new CpsFlowDefinition("pipeline {\n"
+                + "    agent { label 'docker' }\n"
+                + "    stages {\n"
+                + "        stage ('Record Issues') {\n"
+                + "            steps {\n"
+                + "                recordIssues(tools: [taskScanner(highTags: 'FIXMEE', normalTags: 'TODOO', includePattern: '**/*.java')])\n"
+                + "            }\n"
+                + "        }\n"
+                + "    }\n"
+                + "}", true));
+
+        copySingleFileToAgentWorkspace(agent, job, "docker/Main.java", "Main.java");
+
+        AnalysisResult result = scheduleBuildAndAssertStatus(job, Result.SUCCESS);
+
+        assertThat(result.getErrorMessages()).isEmpty();
+        assertThat(result.getInfoMessages()).contains(
+                "Found a total of 2 open tasks",
+                "-> FIXMEE: 1 open tasks",
+                "-> TODOO: 1 open tasks"
+        );
+    }
+
+    /**
+     * Checks if the Analysis result is correctly, when running inside a gcc make container.
+     *
+     * @throws Exception
+     *         When slave creation or waiting for node start fails.
+     */
+    @Test
+    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
+    public void shouldRecordWithGccAndMake() throws Exception {
+        DumbSlave agent = createAgent(gccDockerRule.get());
+
+        WorkflowJob job = createPipeline();
+        job.setDefinition(new CpsFlowDefinition("pipeline {\n"
+                + "    agent { label 'docker' }\n"
+                + "    stages {\n"
+                + "        stage ('build') {\n"
+                + "            steps {\n"
+                + "                sh 'make compile'"
+                + "            }\n"
+                + "        }\n"
+                + "        stage ('Record Issues') {\n"
+                + "            steps {\n"
+                + "                recordIssues(tools: [taskScanner(highTags: 'FIXMEE', normalTags: 'TODOO', includePattern: '**/*.cpp')])\n"
```

Comment:
It would make more sense to check for actual results of the gcc step. Otherwise a simple existing container would be sufficient.